lib/packages/account_addresses/api/account_addresses_api_hive.dart

```diff
@@ -0,0 +1,208 @@
+import 'dart:async';
+import 'package:hive/hive.dart';
+import 'package:hive_flutter/hive_flutter.dart';
+import 'package:komodo_dex/packages/account_addresses/api/account_addresses_api_interface.dart';
+import 'package:komodo_dex/packages/account_addresses/models/wallet_address.dart';
+import 'package:meta/meta.dart';
+
+class HiveException implements Exception {
```

Comment:
The code in this file is great. Only feedback I have is that if we are making the effort to use an abstract interface so that we are not tied to a specific implementation, we can't reference implementation-specific exceptions unless we need to use a specific Exception instead of a general exception and that exception is specific to the implementation. E.g. A valid example would be if we are storing data in the user's hard-drive and we don't have the required permissions.

Reply:
https://github.com/KomodoPlatform/atomicdex-mobile/pull/41/commits/3ea1e55cffd598cfd6ee32b5ffad976a4ef75983